### ISSUE TITLE
Gnome 2.30: JS ERROR: Exception in callback for signal: timer-updated: TypeError: this.actor.clutter_text.set_markup is not a function

### ISSUE THREAD

**Issue by szpak** [codito/gnome-pomodoro#386]
After upgrade to Fedora 29 (Gnome 2.30) I see the following errors in the system log every second.

gnome-shell-extension-pomodoro-0.13.4-5.fc29.x86_64

```
Nov 13 15:59:52 foobar gnome-shell[7403]: Object St.Label (0x55b431399440), has been already deallocated — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such 
as destroy(), dispose(), or remove() vfuncs.
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: == Stack trace for context 0x55b42eea0200 ==
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #0   55b432390c28 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:274 (7f6ce95e6550 @ 43)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #1   7ffdaddbeb80 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #2   7ffdaddbec70 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #3   55b432390b68 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266 (7f6ce95e6430 @ 343)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #4   7ffdaddbfbf0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #5   7ffdaddbfcb0 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #6   7ffdaddbfda0 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #7   55b432390ad0 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116 (7f6ce95dc430 @ 190)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #8   7ffdaddc0d30 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #9   7ffdaddc0e10 b   self-hosted:981 (7f6ceb0edf70 @ 474)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: == Stack trace for context 0x55b42eea0200 ==
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #0   55b432390d60 i   resource:///org/gnome/shell/ui/messageList.js:123 (7f6ceafac820 @ 72)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #1   7ffdaddbcc80 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #2   55b432390cc0 i   resource:///org/gnome/shell/ui/messageList.js:393 (7f6ceafb1430 @ 88)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #3   7ffdaddbdc10 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #4   55b432390c28 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:275 (7f6ce95e6550 @ 81)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #5   7ffdaddbeb80 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #6   7ffdaddbec70 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #7   55b432390b68 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266 (7f6ce95e6430 @ 343)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #8   7ffdaddbfbf0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #9   7ffdaddbfcb0 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #10   7ffdaddbfda0 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #11   55b432390ad0 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116 (7f6ce95dc430 @ 190)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #12   7ffdaddc0d30 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #13   7ffdaddc0e10 b   self-hosted:981 (7f6ceb0edf70 @ 474)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: == Stack trace for context 0x55b42eea0200 ==
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #0   55b432390c28 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:274 (7f6ce95e6550 @ 43)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #1   7ffdaddbeb80 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #2   7ffdaddbec70 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #3   55b432390b68 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266 (7f6ce95e6430 @ 343)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #4   7ffdaddbfbf0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #5   7ffdaddbfcb0 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #6   7ffdaddbfda0 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar gnome-shell[7403]: Object St.Label (0x55b431399440), has been already deallocated — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #7   55b432390ad0 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116 (7f6ce95dc430 @ 190)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #8   7ffdaddc0d30 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #9   7ffdaddc0e10 b   self-hosted:981 (7f6ceb0edf70 @ 474)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: == Stack trace for context 0x55b42eea0200 ==
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #0   55b432390d60 i   resource:///org/gnome/shell/ui/messageList.js:123 (7f6ceafac820 @ 72)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #1   7ffdaddbcc80 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #2   55b432390cc0 i   resource:///org/gnome/shell/ui/messageList.js:393 (7f6ceafb1430 @ 88)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #3   7ffdaddbdc10 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #4   55b432390c28 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:275 (7f6ce95e6550 @ 81)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #5   7ffdaddbeb80 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #6   7ffdaddbec70 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #7   55b432390b68 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266 (7f6ce95e6430 @ 343)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #8   7ffdaddbfbf0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #9   7ffdaddbfcb0 b   self-hosted:977 (7f6ceb0edf70 @ 413)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #10   7ffdaddbfda0 b   resource:///org/gnome/gjs/modules/signals.js:128 (7f6ceb0c18b0 @ 386)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #11   55b432390ad0 i   /usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116 (7f6ce95dc430 @ 190)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #12   7ffdaddc0d30 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (7f6ceb0b0b80 @ 71)
Nov 13 15:59:52 foobar org.gnome.Shell.desktop[7403]: #13   7ffdaddc0e10 b   self-hosted:981 (7f6ceb0edf70 @ 474)
Nov 13 15:59:52 foobar gnome-shell[7403]: JS ERROR: Exception in callback for signal: timer-updated: TypeError: this.actor.clutter_text.set_markup is not a function
                                           setMarkup@resource:///org/gnome/shell/ui/messageList.js:123:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           setBody@resource:///org/gnome/shell/ui/messageList.js:393:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           createBanner/onTimerUpdated<@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:275:21
                                           _emit@resource:///org/gnome/gjs/modules/signals.js:128:27
                                           _onTimerUpdate@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           _emit@resource:///org/gnome/gjs/modules/signals.js:128:27
                                           _onPropertiesChanged@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
Nov 13 15:59:52 foobar gnome-shell[7403]: Object St.Label (0x55b4302f0120), has been already deallocated — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
Nov 13 15:59:52 foobar gnome-shell[7403]: Object St.Label (0x55b4302f0120), has been already deallocated — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
Nov 13 15:59:52 foobar gnome-shell[7403]: JS ERROR: Exception in callback for signal: timer-updated: TypeError: this.actor.clutter_text.set_markup is not a function
                                           setMarkup@resource:///org/gnome/shell/ui/messageList.js:123:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           setBody@resource:///org/gnome/shell/ui/messageList.js:393:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           createBanner/onTimerUpdated<@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:275:21
                                           _emit@resource:///org/gnome/gjs/modules/signals.js:128:27
                                           _onTimerUpdate@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/notifications.js:266:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                           _emit@resource:///org/gnome/gjs/modules/signals.js:128:27
                                           _onPropertiesChanged@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/timer.js:116:9
                                           wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
```

**Comment by szpak** [comment]
May be related to #385. I also observe that behavior in Fedora 29.

**Comment by junrrein** [comment]
This may have been fixed in version 0.14, which was just uploaded to the Fedora 29 repo. Could you check that?

**Comment by szpak** [comment]
I've been using 0.14 in Fedora for a few days and I haven't seen that error. Closing.
